commit 31fdb4ab1b72fa39290931f6a8934eda4eb4f636
Author: Christoph Feck <christoph@maxiom.de>
Date:   Fri Aug 16 14:31:04 2013 +0200

    Fix crash when failing to get audio CD block device
    
    For the udisks2 backend, this is already worked
    around in Solid, but could fail for other reasons.
    
    BUG: 314544
    FIXED-IN: 4.11.1
    REVIEW: 111626

diff --git a/kfile/kfileplacesitem.cpp b/kfile/kfileplacesitem.cpp
index 5a12486..1cb0fa5 100644
--- a/kfile/kfileplacesitem.cpp
+++ b/kfile/kfileplacesitem.cpp
@@ -185,8 +185,15 @@ QVariant KFilePlacesItem::deviceData(int role) const
             if (m_access) {
                 return QUrl(KUrl(m_access->filePath()));
             } else if (m_disc && (m_disc->availableContent() & Solid::OpticalDisc::Audio)!=0) {
-                QString device = d.as<Solid::Block>()->device();
-                return QUrl(QString("audiocd:/?device=%1").arg(device));
+                Solid::Block *block = d.as<Solid::Block>();
+                if (block) {
+                    QString device = block->device();
+                    return QUrl(QString("audiocd:/?device=%1").arg(device));
+                }
+                // We failed to get the block device. Assume audiocd:/ can
+                // figure it out, but cannot handle multiple disc drives.
+                // See https://bugs.kde.org/show_bug.cgi?id=314544#c40
+                return QUrl(QString("audiocd:/"));
             } else if (m_mtp) {
                 return QUrl(QString("mtp:udi=%1").arg(d.udi()));
             } else {
